ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -371,13 +373,23 @@ static boolean isBlobExpired(CloudBlobMetadata metadata, long currentTimeStamp)
    */
   @Override
   public void put(MessageWriteSet messageSetToWrite) throws StoreException {
-    // TODO: Remove the duplicate code by calling putAsync() method and wait on it once we verify that upload works correctly.
     checkStarted();
     if (messageSetToWrite.getMessageSetInfo().isEmpty()) {
       throw new IllegalArgumentException("Message write set cannot be empty");
     }
     checkDuplicates(messageSetToWrite.getMessageSetInfo());
 
+    if (cloudConfig.ambryBackupVersion.equals(CloudConfig.AMBRY_BACKUP_VERSION_2) && !cloudConfig.vcrRequireEncryption) {
```

Comment:
Do we see performance or memory difference by bypassing the CloudWriteChannel?

Reply:
I will have some numbers by the evening.

Reply:
No change in perf, but slight decrease in overall memory usage. Around 4GB.

Reply:
In this case, we don't need this bypass code.